### DISCUSSION TITLE
Deprecated notices for PHP 8.1.6

### DISCUSSION THREAD

**Discussion by taytus** [spatie/http-status-check#90] [General]
I love this package and it is working perfectly fine.

I just saw a couple of notices that I don't believe affect the package but I'm sharing them here just in case:

`PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1789

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1789
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1800

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1800
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1812

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1812
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1827

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1827
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1768

Deprecated: Return type of Tightenco\Collect\Support\Collection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1768
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1747

Deprecated: Return type of Tightenco\Collect\Support\Collection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1747
PHP Deprecated:  Return type of Tightenco\Collect\Support\Collection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1716

Deprecated: Return type of Tightenco\Collect\Support\Collection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1716
Start scanning http://roboamp.com

PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 71

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 71
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 57

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 57
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 62

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 62
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 76

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 76
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 52

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 52
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 81

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 81
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 86

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 86
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 91

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 91
PHP Deprecated:  Return type of Spatie\Crawler\CrawlObserverCollection::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 96

Deprecated: Return type of Spatie\Crawler\CrawlObserverCollection::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/taytus/.composer/vendor/spatie/crawler/src/CrawlObserverCollection.php on line 96`

**Comment by freekmurze** [comment]
I believe you're using an older version of the package.

Feel free to PR improvements.

**Comment by taytus** [comment]
Thank you @freekmurze.

I'm using 3.4. If I'm not mistaken, that's the latest one, right?

**Reply by freekmurze** [reply]
Ah right, that's indeed the latest one.
